src/services/api-rest/identity-resolution/identity-resolution.service.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  Inject,
+  Injectable,
+  InternalServerErrorException,
+  LoggerService,
+  ServiceUnavailableException,
+} from '@nestjs/common';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { UserLookupService } from '@domain/community/user-lookup/user.lookup.service';
+import { RegistrationService } from '@services/api/registration/registration.service';
+import { UserService } from '@domain/community/user/user.service';
+import { KratosService } from '@services/infrastructure/kratos/kratos.service';
+import { AuthenticationService } from '@core/authentication/authentication.service';
+import { randomUUID } from 'node:crypto';
+import { LogContext } from '@common/enums';
+import { UserIdentityNotFoundException } from '@common/exceptions/user/user.identity.not.found.exception';
+import { DuplicateAuthIdException } from '@common/exceptions/user/duplicate.authid.exception';
+import type { Identity } from '@ory/kratos-client';
+import type { OryDefaultIdentitySchema } from '@services/infrastructure/kratos/types/ory.default.identity.schema';
+
+export interface IdentityResolutionResult {
```

Comment:
Please split out interfaces / objects into separate files, not multiple in the same file